api/rpc/perms/permissions.go

```diff
@@ -28,9 +32,16 @@ func (j *JWTPayload) MarshalBinary() (data []byte, err error) {
 
 // NewTokenWithPerms generates and signs a new JWT token with the given secret
 // and given permissions.
-func NewTokenWithPerms(signer jwt.Signer, perms []auth.Permission) ([]byte, error) {
+func NewTokenWithPerms(signer jwt.Signer, perms []auth.Permission, ttl time.Duration) ([]byte, error) {
+	var nonce [32]byte
+	if _, err := rand.Read(nonce[:]); err != nil {
+		return nil, err
+	}
+
 	p := &JWTPayload{
-		Allow: perms,
+		Allow:     perms,
+		Nonce:     nonce[:],
+		ExpiresAt: time.Now().UTC().Add(ttl),
```

Comment:
`.UTC()` to prevent timezone issues.

Reply:
@cristaloleg let's please only add a TTL if one is specified. I think we should preserve the default behaviour which is token is eternal.

Reply:
Done